blobs.go

```diff
@@ -75,6 +75,17 @@ type BlobDeleter interface {
 	Delete(ctx context.Context, dgst digest.Digest) error
 }
 
+// BlobEnumerator allows to list blobs in storage.
+type BlobEnumerator interface {
+	// Enumerate fills 'digests' with a lexicographically sorted list of
+	// digests up to the size of 'digests' and returns the value 'n' for the
+	// number of entries which were filled. 'last' contains an offset in the
+	// list. It's a string representation of digest. Empty string means 0
+	// offset. 'err' will be set to io.OEF if there are no more entries to
+	// obtain.
+	Enumerate(ctx context.Context, digests []digest.Digest, last string) (n int, err error)
```

Comment:
I was thinking something like this:

```
func Enumerate(ctx context.Context, func(dgst digest.Digest) error {
    // Stop if io.EOF returned
})
```

The only issue with this is that it makes pagination somewhat challenging to integrate. I'm not sure if that is a concern for this use case.